service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/credential/mfa/shiro/MfaOptionServiceImpl.java

```diff
@@ -212,7 +208,8 @@ public ScratchCodeListResult createAllScratchCodes(TxContext tx, ScratchCodeCrea
             // Create code
             ScratchCodeImpl codeImpl = new ScratchCodeImpl(scratchCodeCreator.getScopeId(), scratchCodeCreator.getMfaOptionId(), encryptedCode);
 
-            ScratchCode scratchCode = scratchCodeRepository.create(tx, codeImpl);
+            ScratchCode scratchCode = txManager.execute(tx -> scratchCodeRepository.create(tx, codeImpl));
+            scratchCode.setCode(code);
```

Comment:
Please add a comment inline on why we are doing this `setCode` after the creation

Reply:
Thanks for the feedback, just updated the changes 😁